File: RedSheet.DbEntities/Models/ProjectNegotionality.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("ProjectNegotionalities", Schema = "dbo")]
	[Serializable]
	public partial class ProjectNegotionality  
	{
 
	[Required]
	public bool IsMarketDifficult { get; set; } 
 
	[Required]
	public bool IsSpendLarge { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int ProjectNegotionalityId { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public NegotionalityCategory NegotionalityCategoryId { get; set; } 
 
	[RelationshipTableAttribue("ProjectModules","dbo")]
	//Relationships
	public int ProjectModuleId { get; set; } 
 
	[ForeignKey("ProjectModuleId")]
	public ProjectModule ProjectModule { get; set; } 
	[InverseProperty("ProjectNegotionality")]
	public ICollection<OurTeamMember> OurTeamMembers { get; set; } 


	public ProjectNegotionality()
	{
	this.OurTeamMembers = new HashSet<OurTeamMember>();

    }

    }
}

File: redsheet-portal/src/app/database-models/project-negotionality.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule, OurTeamMember,  vProjectNegotionalityRecord  } from './'
export class ProjectNegotionality {
    constructor(projectNegotionality?: ProjectNegotionality  | vProjectNegotionalityRecord )  {
        let properties = [ "isMarketDifficult", "isSpendLarge", "projectNegotionalityId", "negotionalityCategoryId", "projectModuleId", "ourTeamMembers",];
        for (let property of properties)
            if (projectNegotionality && projectNegotionality[property])
                this[property] = projectNegotionality[property];
    }
 
	isMarketDifficult : boolean = false ;
 
	isSpendLarge : boolean = false ;
 
	projectNegotionalityId : number =   0 ;
 
    @range(0,2147483647)
	negotionalityCategoryId : number =   undefined;
 
    @range(0,2147483647)
	projectModuleId : number =   undefined;
	projectModule : ProjectModule  ;
	@nested(OurTeamMember)
	ourTeamMembers: OurTeamMember[];



}
